### ISSUE TITLE
Trying to access array offset on value of type null when not logged in

### ISSUE THREAD

**Issue by tretkowski** [pixelfed/pixelfed#4919]
Since upgrading to the Docker images of 0.11.11 I'm getting an error when trying to access posts and not being logged in. Here is an example:
https://pixel.gruene.social/p/norbert/388986170790350902

Works perfectly fine when logged in.

```
ErrorException:
Trying to access array offset on value of type null

  at /var/www/storage/framework/views/dfb8a4ac0b1b857c2efdea975769fb27.php:8
  at Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:254)
  at Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
     (/var/www/storage/framework/views/dfb8a4ac0b1b857c2efdea975769fb27.php:8)
  at require('/var/www/storage/framework/views/dfb8a4ac0b1b857c2efdea975769fb27.php')
     (/var/www/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:124)
  at Illuminate\Filesystem\Filesystem::Illuminate\Filesystem\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:125)
  at Illuminate\Filesystem\Filesystem->getRequire()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/Engines/PhpEngine.php:58)
  at Illuminate\View\Engines\PhpEngine->evaluatePath()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php:72)
  at Illuminate\View\Engines\CompilerEngine->get()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/View.php:207)
  at Illuminate\View\View->getContents()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/View.php:190)
  at Illuminate\View\View->renderContents()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/View.php:159)
  at Illuminate\View\View->render()
     (/var/www/vendor/laravel/framework/src/Illuminate/Http/Response.php:69)
  at Illuminate\Http\Response->setContent()
     (/var/www/vendor/laravel/framework/src/Illuminate/Http/Response.php:35)
  at Illuminate\Http\Response->__construct()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:911)
  at Illuminate\Routing\Router::toResponse()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:878)
  at Illuminate\Routing\Router->prepareResponse()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:799)
  at Illuminate\Routing\Router->Illuminate\Routing\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:141)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/app/Http/Middleware/Localization.php:21)
  at App\Http\Middleware\Localization->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/app/Http/Middleware/TwoFactorAuth.php:33)
  at App\Http\Middleware\TwoFactorAuth->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/app/Http/Middleware/EmailVerificationCheck.php:35)
  at App\Http\Middleware\EmailVerificationCheck->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php:63)
  at Laravel\Passport\Http\Middleware\CreateFreshApiToken->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:50)
  at Illuminate\Routing\Middleware\SubstituteBindings->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:78)
  at Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php:41)
  at Illuminate\Session\Middleware\AuthenticateSession->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49)
  at Illuminate\View\Middleware\ShareErrorsFromSession->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:121)
  at Illuminate\Session\Middleware\StartSession->handleStatefulRequest()
     (/var/www/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:64)
  at Illuminate\Session\Middleware\StartSession->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37)
  at Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/app/Http/Middleware/FrameGuard.php:18)
  at App\Http\Middleware\FrameGuard->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:67)
  at Illuminate\Cookie\Middleware\EncryptCookies->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:116)
  at Illuminate\Pipeline\Pipeline->then()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:800)
  at Illuminate\Routing\Router->runRouteWithinStack()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:777)
  at Illuminate\Routing\Router->runRoute()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:741)
  at Illuminate\Routing\Router->dispatchToRoute()
     (/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php:730)
  at Illuminate\Routing\Router->dispatch()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:200)
  at Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:141)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php:49)
  at Illuminate\Http\Middleware\HandleCors->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php:39)
  at Illuminate\Http\Middleware\TrustProxies->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
  at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php:31)
  at Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
  at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php:40)
  at Illuminate\Foundation\Http\Middleware\TrimStrings->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php:27)
  at Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php:99)
  at Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:180)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:116)
  at Illuminate\Pipeline\Pipeline->then()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:175)
  at Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
     (/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:144)
  at Illuminate\Foundation\Http\Kernel->handle()
     (/var/www/public/index.php:57)
```

**Comment by eruraindil** [comment]
I am seeing this on my instance, but only on older Instagram imported statuses.

**Comment by eruraindil** [comment]
I've done some digging. Is your status unlisted by chance?  For me it's related to this: https://github.com/pixelfed/pixelfed/commit/fd44c80ce9723cf91ece4767539f0c503b2f833f#diff-138817b942353fe34a3f805cf69f1d063aa679b37906c88f498802996b0caadfR7 

In order to get metadata the status is loaded through StatusService get which is not returning non public statuses by default.

I believe this is a bug, unlisted statuses should still show up while not logged in, they are just unlisted from public feeds, right?

**Comment by tretkowski** [comment]
@eruraindil You're right, it happens on unlisted posts only.

**Comment by dansup** [comment]
Fixed in 0f3ca194
